compiler/optimizer/Operation.cpp

```diff
@@ -0,0 +1,347 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+// #include "optimizer/Operation.hpp"
+#include "optimizer/TreeInterpreter.hpp"
+#include <cstring>
+
+void
+TR::TreeInterpreter::performOp(TR::Node * node){
```

Comment:
We thought that it would be good to separate the responsibilities of the files, and let this one handle just opcodes. Particularly, we were thinking ahead to when the rest of long operations are included, as well as the operations for byte, short, integer, etc.

Let me know your thoughts.